### ISSUE TITLE
Please update old peer dependency "@typescript-eslint/parser"

### ISSUE THREAD

**Issue by tmtron** [nrwl/nx#3715]
## Current Behavior
update to nx 10.0.2 fails:
```
Package "@nrwl/eslint-plugin-nx" has an incompatible peer dependency to "@typescript-eslint/parser" (req
uires "^2.19.2", would install "4.1.0")
```

## Expected Behavior
nx should use the newest version of the related dependencies

## Steps to Reproduce
* In the project `package.json`, depend on a newer version of `@typescript-eslint/eslint-plugin`
```
"devDependencies": {
  "@typescript-eslint/eslint-plugin": "^4.1.0",
```
* then try to update nx: `ng update @nrwl/workspace`

### Failure Logs
```
Package "@nrwl/eslint-plugin-nx" has an incompatible peer dependency to "@typescript-eslint/parser" (req
uires "^2.19.2", would install "4.1.0")
× Migration failed: Incompatible peer dependencies found.
Peer dependency warnings when installing dependencies means that those dependencies might not work correctly together.
You can use the '--force' option to ignore incompatible peer dependencies and instead address these warnings later.
```

### Misc
* as a workaround we can make the update work with the `--force` flag: `ng update @nrwl/workspace --force`
* maybe you also need to update related dependencies, e.g. `@typescript-eslint/eslint-plugin`, `@typescript-eslint/experimental-utils`, ...

### Environment
```
$ nx report

>  NX  Report complete - copy this into the issue template

  nx : Not Found
  @nrwl/angular : 10.0.13
  @nrwl/cli : 10.0.13
  @nrwl/cypress : 10.0.13
  @nrwl/eslint-plugin-nx : 10.0.13
  @nrwl/express : Not Found
  @nrwl/jest : 10.0.13
  @nrwl/linter : 10.0.13
  @nrwl/nest : 10.0.13
  @nrwl/next : Not Found
  @nrwl/node : 10.0.13
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 10.0.13
  @nrwl/web : Not Found
  @nrwl/workspace : 10.0.13
  typescript : 3.9.7

```

**Comment by JamesHenry** [comment]
This will be addressed in https://github.com/nrwl/nx/pull/3763

**Comment by FrozenPandaz** [comment]
Closing this issue as it is fixed in `master` and will be released in `10.3.0` :tada:
